contrib/dunst_xr_theme_changer.sh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bash
+###############################################################################
+#
+# This script creates a dunst user configuration file in
+# $XDG_CONFIG_HOME/dunst/ folder, changing dunst basic teeming options
+# according to your current X resources color palette.
+#
+# It gets your current user configuration file
+# -$HOME/.conf/dunst/dunstrc, which is copied from the default
+# config file in /usr/share/dunst/dunstrc if it does not exist- and
+# changes the defined attributes -declared further on in the script as
+# $theme_attr_dict dictionary- to the new ones. Then it dumps the new
+# configuration to $user_xr_color_conf file.
+#
+###############################################################################
+
+set -e
+
+script_name="$(basename "$0")"
+
+# Function to get resource values
+xrdb_get () {
+    output="$(xrdb -q | grep -e "$1" | cut -f2)"
+    default="$2"
+    printf '%s' "${output:-$default}"
+}
+
+# User config dir and file
+user_conf_dir="${XDG_CONFIG_HOME:-$HOME/.config}/dunst"
+user_conf="$user_conf_dir/dunstrc"
+
+# Default config dir and example file
+example_conf_dir="/usr/share/dunst"
+example_conf="$example_conf_dir/dunstrc"
+
+# User xresources color config file
+user_xr_color_conf="$user_conf_dir/dunstrc_xr_colors"
+
+# Check if the user config directory exists
+if ! [ -d "$user_conf_dir" ]; then
+    printf 'Creating folder "%s".\n' "$user_conf_dir"
+    mkdir -p  "$user_conf_dir"
+fi
+
+# Check if the user config file exists
+if ! [ -f "$user_conf" ]; then
+    printf '"%s" file does not exist.\nChecking for config file example.' \
+               "$user_conf"
+
+    if [ -d "/usr/share/dunst" ]; then
+        # Archlinux default dir and example file
+        example_conf_dir="/usr/share/dunst"
+
+        if [ -f "$example_conf_dir/dunstrc" ]; then
+            example_conf="$example_conf_dir/dunstrc"
+        else
+            printf 'Could not find the example config file in "%s".
+               \nPlease, change $example_conf variable in the script.' \
+                   "$example_conf_dir"
+            exit 1
+        fi
+
+    elif [ -d "/usr/share/doc/dunst" ]; then
+        # Debian/Ubuntu default dir
+        example_conf_dir="/usr/share/doc/dunst"
+
+        if [ -f "$example_conf_dir/dunstrc.example.gz" ]; then
+            # Ubuntu <= 17.10 and Debian <= 1.2.0-2 example file:
+            example_conf="$example_conf_dir/dunstrc.example.gz"
```

Comment:
This won't work, the file in ubuntu and debian is compressed with gzip  you'll need to uncompress it to use it.